README.md

<!-- README.md is generated from README.Rmd. Please edit that file -->

# sleepIPD

<!-- badges: start -->

<!-- badges: end -->

sleepIPD is an opinionated wrapper around the
[GGIR](https://CRAN.R-project.org/package=GGIR) accelerometer processing
scripts. The wrapper ensures that contributors can reprocess data in
precisely the same way, reducing variations in the data that arise from
different processing decisions. In addition to the reprocessing
functions, sleepIPD also provides a series of helper functions to manage
the reprocessing.

## Installation

This package is not available on CRAN. You can install from
[GitHub](https://github.com/Motivation-and-Behaviour/sleepIPD) with:

``` r
# install.packages("devtools")
devtools::install_github("Motivation-and-Behaviour/sleepIPD")
```

## Documentation

You can find the documentation and getting started guide for sleepIPD
[here](https://motivation-and-behaviour.github.io/sleepIPD/articles/sleepIPD.html).
This also includes the steps for reprocessing if your data are large or
more complex.


NEWS.md
# sleepIPD 0.2.2

# sleepIPD 0.2.1

# sleepIPD 0.2.0

* Added a helper function (`assemble_datasheet`) for collecting and merging the reprocessed data.

# sleepIPD 0.1.2

* Fix the algorithm for the hip-worn measurement of sleep.

# sleepIPD 0.1.1

* Fixed the issue related to hip-worn measurement of sleep.

# sleepIPD 0.1.0

* Bumped to version used for reprocessing

# sleepIPD 0.0.0.9000

* Added a `NEWS.md` file to track changes to the package.
* Added base functionality to the package.
* Created a 'getting started' vignette for contributors.


tests/testthat.R
library(testthat)
library(sleepIPD)

test_check("sleepIPD")


tests/testthat/test-helpers.R
test_that("correct thresholds are returned", {
  # Simple
  expect_equal(
    apply_thresholds("light", "children", "geneactiv", "wrist"),
    51.6
  )
  # Multiple
  expect_equal(
    apply_thresholds(
      "mod", c("children", "adults"),
      "geneactiv", "wrist"
    ),
    c(191.6, 93.2)
  )
})

test_that("Windows file paths are translated", {
  expect_match(translate_filepath("C:\\Documents\\Study Name With Spaces"),
    "C:/Documents/Study Name With Spaces",
    fixed = TRUE
  )
})

test_that("study names are validated", {
  expect_match(translate_studyname("A study name with spaces"),
    "A-study-name-with-spaces",
    fixed = TRUE
  )
})


R/helpers.R
#' Choose the correct intensity thresholds
#'
#' A helper function for choosing intensity thresholds. Not intended to be
#' invoked directly by the user.
#'
#' @param intensity The thresholds to return. Must be one of 'light', 'mod', or
#' 'vig'
#' @param ages The age group (i.e., children or adults or both)
#' @param device The device (i.e., actigraph or geneactiv or both)
#' @param wear_location The wear location of the device (i.e., 'wrist' or 'hip')
#' @family helper functions (internal)
#'
#' @importFrom rlang .data
#'
#' @return vector of threshold cut-points
apply_thresholds <- function(intensity, ages, device, wear_location) {
  thresholds %>%
    dplyr::filter(.data$age %in% ages &
      .data$dev %in% device &
      .data$wear_loc %in% wear_location) %>%
    dplyr::pull(intensity)
}

#' Convert a File or Folder Path
#'
#' GGIR uses some code which does not play nice with some file paths on Windows
#' (e.g., those produced by \code{tempdir()}). This function converts between a
#' file path to the format that works with GGIR.
#'
#' @param path The file or folder path to convert.
#'
#' @return The same file path with \code{\\\\} replaced with \code{/}
#' @export
#' @family helper functions
#'
#' @examples
#' example_path <- tempdir()
#' translate_filepath(example_path)
translate_filepath <- function(path) {
  gsub("\\\\", "/", path)
}

#' Remove White Space from a Study Name
#'
#' @param studyname Name of the study
#' @family helper functions (internal)
#'
#' @return Study name with whitespace replaced with \code{-}
translate_studyname <- function(studyname) {
  gsub("\\s", "-", studyname)
}

#' Collate GGIR Outputs
#'
#' This function simply finds the outputs from GGIR that are required for the
#' project, and copies them to a new folder. If there are multiple copies of
#' part 5 outputs, they are collapsed to a single csv file.
#'
#' @param outputdir the outputdir that was provided in the processing call
#' @param studyname the name of the study. Use the first author's surname if
#' there is no study name.
#' @param verbose If success messages should be displayed
#' @family helper functions
#'
#' @export
collate_outputs <- function(outputdir, studyname, verbose = TRUE) {
  outputdir <- translate_filepath(outputdir)
  studyname <- translate_studyname(studyname)

  # Find all of the files
  # Part 2
  p2files <- list.files(
    path = outputdir, pattern = "part2_daysummary.csv",
    recursive = TRUE
  )

  check_outfiles("2", 1, length(p2files), verbose)

  p2files_renamed <- basename(gsub(
    "part2_daysummary",
    paste0("part2_", studyname),
    p2files
  ))

  p4files <- list.files(
    path = outputdir, pattern = "part4_nightsummary_sleep_cleaned.csv",
    recursive = TRUE
  )

  check_outfiles("4", 1, length(p4files), verbose)

  p4files_renamed <- basename(gsub(
    "part4_nightsummary_sleep_cleaned",
    paste0("part4_", studyname),
    p4files
  ))

  p5files <- list.files(
    path = outputdir, pattern = "part5_daysummary_.*.csv",
    recursive = TRUE
  )

  check_outfiles("5", 1024, length(p5files), verbose)


  # Create the directory
  collate_outdir <- file.path(outputdir, "sleepIPD_output")
  dir.create(collate_outdir, showWarnings = FALSE)

  # Copy files
  file.copy(file.path(outputdir, p2files),
    file.path(collate_outdir, p2files_renamed),
    overwrite = TRUE
  )

  file.copy(file.path(outputdir, p4files),
    file.path(collate_outdir, p4files_renamed),
    overwrite = TRUE
  )

  # Read all part 5 into one data frame
  p5df <- readr::read_csv(file.path(outputdir, p5files),
    id = "p5filename",
    show_col_types = FALSE,
    lazy = FALSE
  )

  p5df <- p5df %>%
    dplyr::mutate(p5filename = basename(.data$p5filename))

  readr::write_csv(p5df,
    file.path(collate_outdir, paste0("part5_", studyname, ".csv")),
    append = FALSE, na = ""
  )

  if (verbose) {
    usethis::ui_done(
      "Collated data in {usethis::ui_path(collate_outdir)}"
    )
  }
}

#' Create Pre-filled Metadata Template
#'
#' @param outputdir The directory where the output files were created.
#' @param studyname The name of the study. Use the first author's surname where
#' there is no study name.
#' @param overwrite If an existing metadata template is found, should it be
#' replaced? Default is `FALSE`.
#' @param wear_location Optionally pre-fill the wear location field of the
#' metadata. If there are multiple wear locations, this is ignored.
#'
#' @family helper functions
#'
#' @export
build_meta <- function(outputdir,
                       studyname,
                       overwrite = FALSE,
                       wear_location = NULL) {
  studyname <- translate_studyname(studyname)
  new_filename <- file.path(outputdir, paste0(studyname, "_metadata.xlsx"))
  template <- system.file("extdata", "sleepIPD_metadata_template.xlsx",
    package = "sleepIPD"
  )

  if (!overwrite && file.exists(new_filename)) {
    err_msg <- "A file already exists at {usethis::ui_path(new_filename)}.
    Use {usethis::ui_code('overwrite = TRUE')} to replace."
    usethis::ui_stop(err_msg)
  }

  # Check we can find the part 2 files
  part2file <- file.path(
    outputdir,
    "sleepIPD_output",
    paste0("part2_", studyname, ".csv")
  )

  if (!file.exists(part2file)) {
    err_msg <- "Could not find part 2 data at {usethis::ui_path(part2file)}."
    usethis::ui_stop(err_msg)
  }

  # Create the data for writing
  unique_files <- readr::read_csv(part2file, show_col_types = FALSE) %>%
    dplyr::distinct(.data$filename) %>%
    dplyr::pull(.data$filename)



  # Add new data in
  file.copy(template, new_filename, overwrite = overwrite)
  wb <- openxlsx::loadWorkbook(new_filename)
  openxlsx::writeData(
    wb = wb, sheet = "Metadata", x = unique_files,
    startCol = 1, startRow = 2
  )
  if (length(wear_location) == 1) {
    loc_out <- rep(
      switch(wear_location,
        "wrist" = 1,
        "hip" = 2
      ),
      length(unique_files)
    )
    openxlsx::writeData(
      wb = wb, sheet = "Metadata", x = loc_out,
      startCol = 3, startRow = 2
    )
  }
  openxlsx::saveWorkbook(wb, new_filename, overwrite = TRUE)


  msg <- "Saved metadata template ({usethis::ui_path(new_filename)})"
  usethis::ui_done(msg)
}

#' Check Number of Files Matches Expectations
#'
#' An internal function. Performs a check for output files to confirm the
#' correct number are found.
#'
#' @param part The GGIR part the files relate to (as a string).
#' @param expect The number of files expected to be found.
#' @param found The number of files actually found.
#' @param verbose Should a success message be displayed?
#'
#' @family helper functions (internal)
check_outfiles <- function(part, expect, found, verbose) {
  files <- if (expect == 1) "file" else "files" # nolint

  if (found == 0) {
    usethis::ui_stop(
      "Could not find part {part} {files}. Did GGIR complete?"
    )
  } else if (found > expect) {
    msg <- paste0(
      "Found {usethis::ui_value(found)} part {part} files ",
      "(expected {usethis::ui_value(expect)}).\n",
      "Does your output directory have previous processing?"
    )
    usethis::ui_stop(msg)
  } else if (verbose) {
    usethis::ui_done("Found {usethis::ui_value(found)} part {part} {files}")
  }
}


#' Find Problem Files
#'
#' A helper function to identify which files failed to process. The heuristic is
#' that files that are in the previous metadata but not in the current metadata
#' are likely to be the files that have issues.
#'
#' If this list is long, don't remove all of the files. Instead, use a binary
#' search (or similar) to find which file or files are the problem.
#'
#' @param outputdir The folder that contains the output of GGIR
#' (one level up from the 'meta' folder).
#' @param previous The last fully completed stage of processing that was
#' completed. This is the name of the folder (e.g., 'ms2.out', not the GGIR
#' part number).
#'
#' @return Invisibly returns a list of files that were missing in the current
#' metadata.
#'
#' @family helper functions
#'
#' @export
#' @examples
#' \dontrun{
#' outputdir <- "C:\\Documents\\My Study\\Output\\output_mystudy"
#' find_problem_files(outputdir, "ms2.out")
#' }
find_problem_files <- function(outputdir,
                               previous = c(
                                 "basic", "ms2.out", "ms3.out", "ms4.out",
                                 "ms5.out"
                               )) {
  folders <- c("basic", "ms2.out", "ms3.out", "ms4.out", "ms5.out")
  previous <- rlang::arg_match(previous, folders)
  current <- folders[[which(folders == previous) + 1]]

  meta_dir <- file.path(outputdir, "meta")

  prev_files <- list.files(file.path(meta_dir, previous), recursive = TRUE)
  curr_files <- list.files(file.path(meta_dir, current), recursive = TRUE)

  if (previous == "basic") {
    prev_files <- substring(prev_files, 6)
  }

  missing <- setdiff(prev_files, curr_files)

  if (length(missing) == 0) {
    usethis::ui_done("No files missing from metadata")
  } else {
    usethis::ui_done("Found {usethis::ui_value(length(missing))} missing files")
    usethis::ui_info("These files are not in previous folder:")
    cat(paste0("\t", missing), sep = "\n")
  }

  invisible(missing)
}


R/validate.R
#' Check GGIR Version
#'
#' Performs a check of the installed version of GGIR. The user is invited to
#' install the correct version if it cannot be located.
#'
#' @param verbose Should a success message be displayed? `TRUE` by default.
#'
#' @family validation functions
#'
#' @export
validate_GGIR <- function(verbose = TRUE) { # nolint
  ggir_ver <- utils::packageVersion("GGIR")
  req_ver <- "2.5.1"

  if (ggir_ver == req_ver) {
    if (verbose) {
      usethis::ui_done(
        "Correct version of GGIR located ({usethis::ui_value(ggir_ver)})."
      )
    }
  } else {
    msg <- "Incorrect version of GGIR installed ({usethis::ui_value(ggir_ver)}).
            Version {usethis::ui_value(req_ver)} is required."
    usethis::ui_oops(msg)

    request <-
      "sleepIPD can install the correct version for you. OK to proceed?"

    resp <- usethis::ui_yeah(request, yes = "Yes", no = "No", shuffle = FALSE)

    if (resp) {
      if (!requireNamespace("devtools", quietly = TRUE)) {
        usethis::ui_info(
          paste0(
            "{usethis::ui_code('devtools')} is also required ",
            "and will be installed."
          )
        )
        utils::install.packages("devtools")
      }
      detach("package:GGIR", unload = TRUE)
      devtools::install_github("wadpac/GGIR@2.5-1")
      validate_GGIR()
    } else {
      stop_msg <-
        "Processing cannot be completed without the correct version of GGIR."
      usethis::ui_stop(stop_msg)
    }
  }
}


R/processing.R
#' Process files to SleepIPD specs
#'
#' `process_files` collects information about your study sample and reprocesses
#' the files accordingly.
#'
#' The purpose of this function is to standardise the processing of the
#' accelerometery files. As such, most of the arguments come preset and should
#' not be modified. However, options which do not affect the output (e.g., the
#' decision to process in parallel or not) are available via `...`. See
#' [GGIR::g.shell.GGIR()] for details of these arguments.
#'
#' You can specify parameters of your study by supplying `ages`, `device`,
#' and/or `wear_location` (see Arguments for details). Not supplying these will
#' cause sleepIPD to calculate every combination of thresholds (resulting in
#' \eqn{8^3 \times 2 = 1024} output files). So, supplying them should slightly
#' reduce processing time.
#'
#' @param datadir The directory with the files to process or a vector of
#' file paths to process (as per [GGIR::g.shell.GGIR()]).
#' @param outputdir the directory to store the output (as per
#' [GGIR::g.shell.GGIR()]).
#' @param studyname The name of the study. Only used if `datadir` is a vector of
#' files.
#' @param ages The age of the sample. Can be either `"children"` (for those 18
#' or less), `"adults"` (for >18) or both (i.e., `c("children", "adults")`). If
#' you do not specify, both are used.
#' @param device The device used in the study. Can be either `"geneactiv"`,
#'  `"actigraph"` or (in the unlikely event two devices were used) both (i.e.,
#'  `c("geneactiv", "actigraph")`). If you do not specify, both are used.
#' @param wear_location The location the device was warn. Can be either
#' `"wrist"` (the default) or `"hip"` If the study has mixed wear locations,
#' you need to run the function twice, once for each location.
#' @param overwrite If the existing output should be replaced. `FALSE` by
#' default.
#' @param verbose Determines how much output the function returns. Useful for
#' understanding what the package is doing. `TRUE` by default.
#' @param redo For advanced usecases. If `TRUE`, the function will redo parts
#' 2 through 5, assuming that part 1 has already been completed.
#' @param ... Pass additional parameters to [GGIR::g.shell.GGIR()]. Refer to the
#' `GGIR` manual for details.
#'
#' @family processing functions
#'
#' @export
process_files <- function(datadir,
                          outputdir,
                          studyname = "NA",
                          ages = c("children", "adults"),
                          device = c("geneactiv", "actigraph"),
                          wear_location = "wrist",
                          overwrite = FALSE,
                          verbose = TRUE,
                          redo = FALSE,
                          ...) {
  validate_GGIR(verbose)


  # Build the cut point settings
  ages <- match.arg(ages, c("children", "adults"), several.ok = TRUE)
  device <- match.arg(device, c("geneactiv", "actigraph"), several.ok = TRUE)
  wear_location <- match.arg(wear_location, c("wrist", "hip"))
  sleep_window <- if (wear_location == "hip") "TimeInBed" else "SPT"
  HASPT <- if (wear_location == "hip") "HorAngle" else "HDCZA"

  threshold_lig <- apply_thresholds("light", ages, device, wear_location)
  threshold_mod <- apply_thresholds("mod", ages, device, wear_location)
  threshold_vig <- apply_thresholds("vig", ages, device, wear_location)
  if (verbose) usethis::ui_done("Cutpoints set")

  # Check file paths
  datadir <- translate_filepath(datadir)
  outputdir <- translate_filepath(outputdir)

  # Change settings for redo
  mode <- if (redo) c(2, 3, 4, 5) else c(1, 2, 3, 4, 5)

  # Call GGIR
  if (verbose) usethis::ui_info("Calling GGIR")

  GGIR::g.shell.GGIR(
    mode = mode,
    datadir = datadir,
    outputdir = outputdir,
    do.report = c(2, 4, 5),
    overwrite = overwrite,
    do.enmo = TRUE,
    do.hfen = TRUE,
    do.enmoa = TRUE,
    studyname = studyname,
    # =====================
    # Part 2
    # =====================
    do.imp = TRUE,
    strategy = 2,
    maxdur = 0,
    includedaycrit = 16,
    qwindow = c(0, 6, 12, 18, 24),
    mvpathreshold = threshold_mod,
    bout.metric = 4,
    excludefirstlast = FALSE,
    includenightcrit = 16,
    MX.ig.min.dur = 10,
    iglevels = 50,
    ilevels = seq(0, 4000, 50),
    epochvalues2csv = FALSE,
    winhr = c(
      16, # Most active 16 hours
      c(60 / 60), # Most active hour
      c(30 / 60), # Most active half hour
      c(15 / 60), # Most active 15 min
      c(10 / 60), # Most active 10 min
      c(5 / 60) # Most active 5 min)
    ),
    qlevels = c(
      960 / 1440, # 2/3rds of the day
      1320 / 1440, # Top 120min
      1380 / 1440, # Top 60min
      1410 / 1440, # Top 30min
      1425 / 1440, # Top 15min
      1435 / 1440 # Top 5min
    ),
    # =====================
    # Part 3 + 4
    # =====================
    def.noc.sleep = 1,
    outliers.only = TRUE,
    criterror = 4,
    do.visual = TRUE,
    sensor.location = wear_location,
    sleepwindowType = sleep_window,
    HASPT.algo = HASPT,
    # =====================
    # Part 5
    # =====================
    threshold.lig = threshold_lig,
    threshold.mod = threshold_mod,
    threshold.vig = threshold_vig,
    boutcriter = 0.8, boutcriter.in = 0.9, boutcriter.lig = 0.8,
    boutcriter.mvpa = 0.8, boutdur.in = c(1, 10, 30), boutdur.lig = c(1, 10),
    boutdur.mvpa = c(1),
    includedaycrit.part5 = 2 / 3,
    # =====================
    # Visual report
    # =====================
    timewindow = c("WW", "MM"),
    visualreport = FALSE,
    ...
  )
}

#' Reprocess Accelerometer Data and Generate Metadata
#'
#' This is the primary wrapper function for sleepIPD. In most cases, this should
#' be the only function that is needed. `reprocess` collects data about your
#' study, and then calls the functions to reprocess the data, collate the
#' results, and generate the metadata.
#'
#' The purpose of sleepIPD is to standardise the processing of the
#' accelerometery files. As such, most of the arguments come preset and should
#' not be modified. However, options which do not affect the output (e.g., the
#' decision to process in parallel or not) are available via `...`. See
#' [GGIR::g.shell.GGIR()] for details of these arguments.
#'
#' You can specify parameters of your study by supplying `ages`, `device`,
#' and/or `wear_location` (see Arguments for details). Not supplying these will
#' cause sleepIPD to calculate every combination of thresholds (resulting in
#' \eqn{8^3 \times 2 = 1024} output files). So, supplying them should slightly
#' reduce processing time.
#'
#' You should also supply a `studyname` argument. This is required by GGIR if
#' your `datadir` is a vector of files, but is additionally used by sleepIPD to
#' name the output files. If your study did not have a name, the last name of
#' the first author is sufficent (if you are supplying multiple studies, name
#' each one with the year the data were collected; e.g.,
#' \code{studyname = 'Hartwig 2021'}).
#'
#' @param datadir The directory with the files to process or a vector of
#' file paths to process (as per [GGIR::g.shell.GGIR()]).
#' @param outputdir the directory to store the output (as per
#' [GGIR::g.shell.GGIR()]).
#' @param studyname The name of the study. This is required to name the
#' metadata.
#' @param ages The age of the sample. Can be either `"children"` (for those 18
#' or less), `"adults"` (for >18) or both (i.e., `c("children", "adults")`). If
#' you do not specify, both are used.
#' @param device The device used in the study. Can be either `"geneactiv"`,
#'  `"actigraph"` or (in the unlikely event two devices were used) both (i.e.,
#'  `c("geneactiv", "actigraph")`). If you do not specify, both are used.
#' @param wear_location The location the device was warn. Can be either
#' `"wrist"` (the default) or `"hip"` If the study has mixed wear locations,
#' you need to run the function twice, once for each location.
#' @param overwrite If the existing output (including metadata) should be
#' replaced. `FALSE` by default.
#' @param verbose Determines how much output the function returns. Useful for
#' understanding what the package is doing. `TRUE` by default.
#' @param ... Pass additional parameters to [GGIR::g.shell.GGIR()]. Refer to the
#' `GGIR` manual for details.
#'
#' @family processing functions
#' @export
#'
#' @examples
#' \dontrun{
#' datadir <- "C:\\Documents\\My Study\\Data"
#' outputdir <- "C:\\Documents\\My Study\\Output"
#' studyname <- "Hartwig 2021"
#' ages <- "children"
#' device <- "geneactiv"
#' wear_location <- "wrist"
#'
#' reprocess(
#'   datadir = datadir,
#'   outputdir = outputdir,
#'   studyname = studyname,
#'   ages = ages,
#'   device = device,
#'   wear_location = wear_location
#' )
#' }
reprocess <- function(datadir,
                      outputdir,
                      studyname,
                      ages = c("children", "adults"),
                      device = c("geneactiv", "actigraph"),
                      wear_location = "wrist",
                      overwrite = FALSE,
                      verbose = TRUE,
                      ...) {
  start_time <- Sys.time()
  if (verbose) {
    usethis::ui_info(
      "Starting {usethis::ui_code('process_files()')}"
    )
  }
  process_files(
    datadir, outputdir, studyname, ages, device,
    wear_location, overwrite, verbose, ...
  )
  end_process_files <- Sys.time()

  if (verbose) {
    print("/n")
    usethis::ui_done("Completed {usethis::ui_code('process_files()')}")
    usethis::ui_info("Starting {usethis::ui_code('collate_outputs()')}")
  }

  collate_outputs(outputdir, studyname, verbose)
  end_collate_outputs <- Sys.time()

  if (verbose) {
    usethis::ui_done("Completed {usethis::ui_code('collate_outputs()')}")
    usethis::ui_info("Starting {usethis::ui_code('build_meta()')}")
  }

  build_meta(outputdir, studyname, overwrite, wear_location)
  end_build_meta <- Sys.time()

  if (verbose) usethis::ui_done("Completed {usethis::ui_code('build_meta()')}")

  if (verbose && requireNamespace("knitr", quietly = TRUE)) {
    usethis::ui_done("Processing complete! Here is how long it took:")

    time_str <- function(endtime, starttime) {
      out <- difftime(endtime, starttime)
      paste(round(as.numeric(out), 2), units(out))
    }

    opts <- options(knitr.kable.NA = "-")

    print(
      knitr::kable(data.frame(
        "Process" = c(
          "GGIR Processing", "Collate Outputs",
          "Build Metadata", "Total"
        ),
        "Start Time" = c(
          start_time, end_process_files,
          end_collate_outputs, NA
        ),
        "End Time" = c(
          end_process_files, end_collate_outputs,
          end_build_meta, NA
        ),
        "Duration" = c(
          time_str(end_process_files, start_time),
          time_str(end_collate_outputs, end_process_files),
          time_str(end_build_meta, end_collate_outputs),
          time_str(end_build_meta, start_time)
        )
      ), format = "simple")
    )

    options(opts)
  }
}


data-raw/thresholds.R
# These are the thresholds for the intensity cut-points. These come from
# Hildebrand 2014 (https://doi.org/10.1249/mss.0000000000000289),
# Hildebrand 2016 (https://doi.org/10.1111/sms.12795),
# Hurter 2018 (https://doi.org/10.3390/children5120172) using non-dominant hand

# nolint start
thresholds <-
  dplyr::tribble(
    ~age, ~dev, ~wear_loc, ~light, ~mod, ~vig, ~source,
    "children", "geneactiv", "wrist", 51.6, 191.6, 695.8,
    "Light: [Hurter et al., 2018](https://doi.org/10.3390/children5120172)<br>
     Mod: [Hildebrand et al., 2014](https://doi.org/10.1249/mss.0000000000000289)<br>
     Vig: [Hildebrand et al., 2014](https://doi.org/10.1249/mss.0000000000000289)",
    "children", "actigraph", "wrist", 48.1, 201.4, 707,
    "Light: [Hurter et al., 2018](https://doi.org/10.3390/children5120172)<br>
    Mod: [Hildebrand et al., 2014](https://doi.org/10.1249/mss.0000000000000289)<br>
    Vig: [Hildebrand et al., 2014](https://doi.org/10.1249/mss.0000000000000289)",
    "children", "geneactiv", "hip", 64.1, 152.8, 514.3,
    "Light: [Hildebrand et al., 2016](https://doi.org/10.1111/sms.12795)<br>
    Mod: [Hildebrand et al., 2014](https://doi.org/10.1249/mss.0000000000000289)<br>
    Vig: [Hildebrand et al., 2014](https://doi.org/10.1249/mss.0000000000000289)",
    "children", "actigraph", "hip", 32.6, 142.6, 464.6,
    "Light: [Hurter et al., 2018](https://doi.org/10.3390/children5120172)<br>
    Mod: [Hildebrand et al., 2014](https://doi.org/10.1249/mss.0000000000000289)<br>
    Vig: [Hildebrand et al., 2014](https://doi.org/10.1249/mss.0000000000000289)",
    "adults", "geneactiv", "wrist", 45.8, 93.2, 418.3,
    "Light: [Hildebrand et al., 2016](https://doi.org/10.1111/sms.12795)<br>
    Mod: [Hildebrand et al., 2014](https://doi.org/10.1249/mss.0000000000000289)<br>
    Vig: [Hildebrand et al., 2014](https://doi.org/10.1249/mss.0000000000000289)",
    "adults", "actigraph", "wrist", 44.8, 100.6, 428.8,
    "Light: [Hildebrand et al., 2016](https://doi.org/10.1111/sms.12795)<br>
    Mod: [Hildebrand et al., 2014](https://doi.org/10.1249/mss.0000000000000289)<br>
    Vig: [Hildebrand et al., 2014](https://doi.org/10.1249/mss.0000000000000289)",
    "adults", "geneactiv", "hip", 46.9, 68.7, 266.8,
    "Light: [Hildebrand et al., 2016](https://doi.org/10.1111/sms.12795)<br>
    Mod: [Hildebrand et al., 2014](https://doi.org/10.1249/mss.0000000000000289)<br>
    Vig: [Hildebrand et al., 2014](https://doi.org/10.1249/mss.0000000000000289)",
    "adults", "actigraph", "hip", 47.4, 61.1, 258.7,
    "Light: [Hildebrand et al., 2016](https://doi.org/10.1111/sms.12795)<br>
    Mod: [Hildebrand et al., 2014](https://doi.org/10.1249/mss.0000000000000289)<br>
    Vig: [Hildebrand et al., 2014](https://doi.org/10.1249/mss.0000000000000289)"
  )
# nolint end

usethis::use_data(thresholds, overwrite = TRUE, internal = TRUE)
